xbmc/network/WebServer.cpp

```diff
@@ -113,8 +117,12 @@ int CWebServer::AskForAuthentication(const HTTPRequest& request) const
 
   LogResponse(request, MHD_HTTP_UNAUTHORIZED);
 
-  ret =
-      MHD_queue_basic_auth_fail_response(request.connection, CCompileInfo::GetAppName(), response);
+  // This MHD_RESULT cast is only necessary for libmicrohttpd 0.9.71
+  // The return type of MHD_queue_basic_auth_fail_response was fixed for future versions
+  // See
+  // https://git.gnunet.org/libmicrohttpd.git/commit/?id=860b42e9180da4dcd7e8690a3fcdb4e37e5772c5
+  ret = (MHD_RESULT)MHD_queue_basic_auth_fail_response(request.connection,
```

Comment:
This is still a C-style cast, please change

Reply:
Change to `static_cast<MHD_RESULT>(MHD_queue_basic_auth_fail_response(request.connection, CCompileInfo::GetAppName(), response));`